### ISSUE TITLE
Disconnect from P2P failed transactions

### ISSUE THREAD

**Issue by vncoelho** [neo-project/neo#916]
If the node receives an invalid TX it should disconnect from that peer and blacklist it (a set such as `known_hashes`).
They will ensure that all nodes are updated and are not propagating invalid txs.

**Comment by lock9** [comment]
Is a single transaction is enough to block the node? 
What parameters are using to define "failed transactions"? What happens if we have a "heavy user", like a big exchange, and they end up sending a few invalid transactions due to errors? There should be some kind of white-listing too.

We have to be careful to avoid creating some kind of censorship in the network. If nodes block ip-addresses just because "someone told us they are bad people", we can put decentralization at risk.

Nodes should block these nodes, but I'm not in favor of propagating this blocking to the whole network.

**Comment by vncoelho** [comment]
For example, consensus are not RPC nodes and, in general, tx just from from RPC interfaces.

In this sense, use the P2P to propagate invalid TXs should not be tolerated. Nodes should be upgrade to the last version of stop to rebroadcast that.

It is not censorship, it is a selection of the peers that make you effectively contributing and not losing computational efforts.

**Comment by lock9** [comment]
I agree with disconnecting, and even creating a blacklist, however, nodes should not be able to censor other nodes.
Imagine this case, we have nodes:
`A`, `B`, `C`, `D`, `E` and `F `

It is not fair that `A` tells `B`, `C`, `D` and `E` that `F` is malicious, making everyone block `F` from the network.
Also, what happens when we have multiple ip addresses in the same place? How long this blacklisting is going to last?

Do we need to disconnect peers who send invalid transactions? Yes
Should we add them to a blacklist? Yes
Should we tell other nodes? No

This is useful, I'm just not in favor of other nodes having the "capability" of censorship upon other users.

**Comment by shargon** [comment]
we should implement an `acl` like on neo-sharp

**Comment by lock9** [comment]
@vncoelho  are you ok of not having this list shared between nodes? An ACL is more than enough and more fair.
If you guys agree, I will remove the discussion tag. I'm still thinking about issues that "are ready to be implemented". Not sure how to track  that 🤔
